From 9ca6936c80cf02b37d6bcf2f58ef2bfdac0d52b2 Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Tue, 22 Oct 2013 19:10:06 +0200
Subject: [PATCH 086/179] BUG/MINOR: acl: implicit arguments of ACL keywords
 were not properly resolved

William Lallemand reported a bug which happens when an ACL keyword using an
implicit argument (eg: a proxy name) is used : the keyword is not properly
set in the arglist field, resulting in an error about the previous keyword
being returned, or "(null)" if the faulty ACL appears first.

The bug only affects error reporting and is 1.5-specific, so no backport is
nedeed.
---
 src/acl.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/acl.c b/src/acl.c
index 7dc25fb..89bfdf5 100644
--- a/src/acl.c
+++ b/src/acl.c
@@ -1134,6 +1134,10 @@ struct acl_expr *parse_acl_expr(const char **args, char **err, struct arg_list *
 			expr->args[0].data.str.str = strdup("");
 			expr->args[0].data.str.size = 1;
 			expr->args[0].data.str.len = 0;
+
+			al->ctx = ARGC_ACL;
+			al->kw = expr->kw;
+			al->conv = NULL;
 			arg_list_add(al, &expr->args[0], 0);
 
 			expr->args[1].type = ARGT_STOP;
-- 
1.8.1.5

